aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -707,11 +718,64 @@ public AWSIotMqttManager(String mqttClientId, Region region, String accountEndpo
         this.mqttClientId = mqttClientId;
         this.region = region;
         this.endpoint = null;
-        this.customWebsocketHeaders = new Properties();
 
         initDefaults();
     }
 
+    /**
+     * Constructs a new AWSIotMqttManager. Use this constructor to connect to AWS IoT using your custom
+     * domain. Other overloads of this constructor support only AWS IoT endpoints which are expected to be
+     * in specific formats. This constructor can however be used for both custom domains as well as AWS IoT managed domains.
+     *
+     * @param region The AWS region to use when creating endpoint.
+     * @param mqttClientId MQTT client ID to use with this client.
+     * @param endpoint Customer specific endpoint from AWS IoT or fully qualified domain name of
+     *                 your custom domain.
+     */
+    private AWSIotMqttManager(
+            final Region region,
+            final String mqttClientId,
+            final String endpoint) {
+
+        if (TextUtils.isEmpty(mqttClientId)) {
+            throw new IllegalArgumentException("mqttClientId is null or empty");
+        }
+
+        if (region == null) {
+            throw new IllegalArgumentException("region is null");
+        }
+
+        if (endpoint == null) {
+            throw new IllegalArgumentException("endpoint is null or an invalid domain.");
```

Comment:
Might as well use `TextUtils.isEmpty(...)` here, too, I guess. Either way would update the error message, the "or an invalid domain." part is not actually checked, right?